components/script/dom/bindings/str.rs

```diff
@@ -302,6 +302,26 @@ impl DOMString {
     pub fn is_valid_week_string(&self) -> bool {
         parse_week_string(&*self.0).is_ok()
     }
+
+    /// A valid local date and time string should be {date}T{time}
+    /// where date and time are both valid
+    /// https://html.spec.whatwg.org/multipage/#valid-local-date-and-time-string
+    pub fn is_valid_local_date_and_time_string(&self) -> bool {
+        parse_local_date_and_time_string(&*self.0).is_ok()
+    }
+
+    /// A valid normalized local date and time string should be {date}T{time}
+    /// where date and time are both valid, and the time string must be as short as possible
+    /// https://html.spec.whatwg.org/multipage/#valid-normalised-local-date-and-time-string
+    pub fn convert_valid_normalized_local_date_and_time_string(&mut self) {
+        let ((year, month, day), (hour, minute, second)) = parse_local_date_and_time_string(&*self.0).unwrap();
```

Comment:
Is there a way to remove this unwrap call? Perhaps by making this function not mutate instead, and make it return a Result? As it stands now, this function is left to the programmer to "be smart enough" to validate the string BEFORE calling this function, throwing away our compile-time guarantees.

Reply:
Another idea would be to simply return if the parsing function returns Err.